components/waitwhile/app/waitwhile.app.ts

```diff
@@ -1,13 +1,197 @@
 import { defineApp } from "@pipedream/types";
+import api from "api";
 
 export default defineApp({
   type: "app",
   app: "waitwhile",
-  propDefinitions: {},
+  propDefinitions: {
+    addTag: {
+      label: "Add Tag",
+      type: "string",
+      optional: true,
+      description: "Adds specified tag to existing tags",
+    },
+    customerId: {
```

Comment:
Would be nice to add `async options()` as mentioned in [this comment](https://github.com/PipedreamHQ/pipedream/pull/4224#discussion_r971062765).
This makes user selection easier and enhances the user experience.

Reply:
Can a property have more than one async options?  Theses props are passed to several different functions so I would need an async option for each function.

Reply:
Yes, you can send parameters to async options, please refer to [the documentation](https://pipedream.com/docs/components/guidelines/#async-options) and [this example](https://github.com/PipedreamHQ/pipedream/blob/master/components/trello/actions/add-attachment-to-card-via-url/add-attachment-to-card-via-url.mjs#L22).